qiskit/providers/aer/extensions/snapshot_density_matrix.py

```diff
@@ -0,0 +1,39 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+from qiskit import QuantumCircuit
+from qiskit.circuit.quantumregister import QuantumRegister
+from qiskit.providers.aer.extensions import Snapshot
+
+class SnapshotDensityMatrix(Snapshot):
+    def __init__(self,
+                 label,
+                 snapshot_type='density_matrix',
```

Comment:
I think it would be preferable not to take the `snapshot_type` as a parameter, as you know that it should be `'density_matrix'`. Could you remove this, and just make the call below `super().__init__(label, 'density_matrix', num_qubits, num_clbits, params)` instead? This applies to all the snapshot classes.